### PR TITLE
rust: use python@3.8

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/linuxbrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/linuxbrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
- [ ] Have you included the output of `brew gist-logs <formula>` of the build failure if your PR fixes a build failure. Please quote the exact error message.

-----

### CODE REVIEW

**Comment by dawidd6** [comment]
> Build `starship` if this get merged.

File: Formula/rust.rb [hunk 1]
```diff
@@ -28,18 +28,19 @@ class Rust < Formula
...
-    depends_on "curl"
-    depends_on "zlib"
+  uses_from_macos "binutils"
+  uses_from_macos "curl"
+  uses_from_macos "zlib"
+
+  if OS.mac?
+    uses_from_macos "python@2"
+  else
+    depends_on "python@3.8"
```
Comment (issyl0) [inline Formula/rust.rb]: At a later date, is there a plan to convert this to use Python 3 upstream as well? That feels like a nice thing to do.

Comment (dawidd6) [reply Formula/rust.rb]: Yes, I think so.

Comment (iMichka) [reply Formula/rust.rb]: Yes. I debug stuff in linuxbrew first, as it is easier to see if python 2 is being picked up. I then backport the change to macOS.

**Pull request merged**